internal/txapp/txapp.go

```diff
@@ -63,6 +63,7 @@ type TxApp struct {
 
 	// transaction that exists between Begin and Commit
 	currentTx        sql.OuterTx
+	genesisTx        sql.OuterTx
```

Comment:
Perhaps some docs on the field describing intended use pattern (where set, where checked, etc.) with a link to https://github.com/cometbft/cometbft/issues/203 or the relevant docs that describe the InitChain exception.